frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -190,19 +191,50 @@ interface AddNewRecordProps {
   collections: string[];
   onSubmit(values: AddNewNosqlRecordFormValues): void;
   transformerHandler: TransformerHandler;
+  data: JobMappingFormValues[];
 }
 
 const AddNewNosqlRecordFormValues = Yup.object({
   collection: Yup.string().required(),
-  key: Yup.string().required(),
+  key: Yup.string()
+    .required()
+    .test({
+      name: 'uniqueMapping',
+      message: 'This key already exists in the selected collection.',
+      test: function (value, context) {
+        const { collection } = this.parent;
+
+        if (!collection || !value) {
+          return true;
+        }
+
+        const selectedCollection: JobMappingFormValues[] =
+          context?.options?.context?.data.filter(
+            (item: JobMappingFormValues) =>
+              `${item.schema}.${item.table}` === collection
+          );
+
+        const isDuplicate = selectedCollection.some(
+          (item: JobMappingFormValues) => item.column === value
+        );
```

Comment:
This can all go away and should be replaced by an `isDuplicate` function that I've mentioned throughout this PR.

Reply:
Also, the `useForm` hook can take in a Context interface, which it looks like that was never created for the `AddNewNosqlRecordForm`. Probably worthwhile to create one. You can see a few other examples throughout the codebase for how to create this. Gives us better type safety.